### PR TITLE
TST: Filter false positive matplotlib warning

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).

xref https://github.com/matplotlib/matplotlib/issues/23614.

This false positive warning showed up in our test suite and can be filtered out.

### CODE REVIEW

**Review by phofl — APPROVED** [review]
lgtm

**Comment by phofl** [comment]
thx @mroeschke

File: pandas/plotting/_matplotlib/core.py [hunk 1]
```diff
@@ -1177,7 +1177,11 @@ def _plot_colorbar(self, ax: Axes, **kwds):
         # use the last one which contains the latest information
         # about the ax
         img = ax.collections[-1]
-        return self.fig.colorbar(img, ax=ax, **kwds)
+        with warnings.catch_warnings():
+            # https://github.com/matplotlib/matplotlib/issues/23614
+            # False positive deprecation warning until matplotlib=3.6
+            warnings.filterwarnings("ignore", "Auto-removal of grids")
```
Comment (jbrockmendel) [inline pandas/plotting/_matplotlib/core.py:1183]: is this unnecessary now that our min version is 3.6.1?

Comment (mroeschke) [reply pandas/plotting/_matplotlib/core.py:1183]: Yeah I think based on https://github.com/matplotlib/matplotlib/issues/23614#issuecomment-1231935245 should be safe to remove

**Pull request merged**